src/datasets/utils/_hf_hub_fixes.py

```diff
@@ -60,6 +66,111 @@ def create_repo(
+    repo_url, to `RepoUrl` object. This function checks the huggingface_hub version to get the repo_id from
+    the repo_url, in the correct way.
+
+    Args:
+        repo_url (`str` or `huggingface_hub.hf_api.RepoUrl`): URL to the repo.
+
+    Returns:
+        `str`: repo_id, the ID of the repository to push to in the following format: `<user>/<dataset_name>` or
+            `<org>/<dataset_name>`.
+    """
+    if version.parse(huggingface_hub.__version__) < version.parse("0.12.0"):
+        from urllib.parse import urlparse
+
+        repo_id = urlparse(repo_url).path[:1]
+        return repo_id
+    else:
+        return repo_url.repo_id
+
+
+def create_pr_it_does_not_exist(
+    hf_api: HfApi,
+    repo_id: str,
+    token: Optional[str] = None,
+    private: Optional[bool] = False,
+    repo_type: Optional[str] = "dataset",
+    create_pr: Optional[bool] = False,
+    branch: Optional[str] = None,
+) -> str:
+    """
+    This function creates a PR for a dataset if it does not exist safely. The RepositoryNotFoundError was introduced in
+    huggingface_hub 0.7.0. This function checks the huggingface_hub version to call the right parameters.
+    The `create_pr` parameter was introduced in huggingface_hub 0.9.0. This function checks the huggingface_hub version
+    to call the right parameters.
+
+    Args:
+        hf_api (`huggingface_hub.HfApi`): Hub client
+        repo_id (`str`): The ID of the repository to push to in the following format: `<user>/<dataset_name>` or
+            `<org>/<dataset_name>`.
+        token (`str`, *optional*): user or organization token. Defaults to None.
+        private (`bool`, *optional*):
+            Whether the model repo should be private.
+        repo_type (`str`, *optional*):
+            Set to `"dataset"` or `"space"` if uploading to a dataset or
+            space, `None` or `"model"` if uploading to a model. Default is
+            `None`.
+        create_pr (`bool`, *optional*):
+            Whether to create a PR if the branch does not exist. Defaults to False.
+        branch (`str`, *optional*):
+            The branch to create the PR on. Defaults to None.
+    """
+    # By version huggingface_hub 0.7.0 HTTPError was replaced by RepositoryNotFoundError
+    # so we need to check the version to use the right exception
+    if version.parse(huggingface_hub.__version__) < version.parse("0.7.0"):
+        from requests.exceptions import HTTPError
+
+        repo_not_found_exception = HTTPError
+    else:
+        from huggingface_hub.hf_api import RepositoryNotFoundError
+
+        repo_not_found_exception = RepositoryNotFoundError
+
+    # Try to find the repo before creating it
+    try:
+        hf_api.repo_info(repo_id, token=token)
+
+    except repo_not_found_exception:
+        repo_url = create_repo(
+            hf_api=hf_api, repo_id=repo_id, private=private, token=token, exist_ok=True, repo_type=repo_type
+        )
+        repo_id = get_repo_id_from_repo_url(repo_url)
```

Comment:
I would remove this line because we already have `repo_id` (we just passed it as parameter to the function call above)
```suggestion
```